pkgs/misc/drivers/spdk/default.nix

```diff
@@ -0,0 +1,49 @@
+{ lib
+, stdenv
+, numactl
+, libaio
+, cunit
+, libuuid
+, fetchFromGitHub
+, gcc
+, openssl
+, python
+}:
+
+stdenv.mkDerivation rec {
+  name = "spdk-${version}";
+  version = "v18.04-pre";
+
+  src = fetchFromGitHub {
+    repo = "spdk";
+    owner = "spdk";
+    rev = "a3f8876777762f9b99544b78ee0b24ff66266c5c";
+    sha256 = "1hb22h8c8b3q91ypfj084ic8gw1fibhfckivfgl334hbbxvay0j0";
+    fetchSubmodules = true;
```

Comment:
It would be nice if could avoid this. I'm not sure how the build system works but we do have "dpdk" packaged already - maybe you can get spdk to use it?